src/v2/Apps/Artwork/Components/ArtworkImageBrowser/utilityTypes.ts

```diff
@@ -0,0 +1,18 @@
+import { Override } from "v2/Utils/typeSupport"
+
+type GenericImages<
+  T extends { images: Readonly<Array<any | null>> | null }
+> = NonNullable<T["images"]>
+type GenericImage<
+  T extends { images: Readonly<Array<any | null>> | null }
+> = Override<NonNullable<GenericImages<T>[number]>, { type: "Image" }>
+type GenericVideo<T extends { video: Readonly<any | null> | null }> = Override<
+  NonNullable<T["video"]>,
+  { type: "Video" }
+>
+export type GenericFigure<
+  T extends {
+    images: Readonly<Array<any | null>> | null
+    video: Readonly<any | null> | null
+  }
+> = GenericImage<T> | GenericVideo<T>
```

Comment:
I don't entirely see why we'd need more typing other than what's provided by Metaphysics. Should model a `figure` there rather than here.

Reply:
This isn't more typing, it's parsing targeted information out of what's provided by metaphysics in order to create the Figure type.

Reply:
Then it should be in Metaphysics though?

Reply:
Aside from possibly keeping the types in MP which should eliminate the need to draft our own, it's slightly easier in the long run to keep types close to the implementation _unless_ said types file will eventually grow in size substantially (think, 3rd-party library size). But even then, co-located close to call-sites is generally easier to comprehend since one doesn't need to jump around different files.

Reply:
This set of types was meant only to be used in the 2 different versions of the ArtworkImageBrowser, hence I kept it in the associated folder. Leaving it in one or the other file didn't better and it cluttered things up. This seemed like a happy medium. Either way, they're not longer relevant.

Reply:
Cool, that makes sense and does seem to a reasonable trade-off should this type need to return 👍 

Another thought would be to import it directly from File A into File B, since File A and B are similar enough.